src/dune_engine/sandbox.ml

```diff
@@ -98,8 +134,20 @@ let create ~mode ~deps ~rule_dir ~chdirs ~rule_digest ~expand_aliases =
   in
   (* CR-someday amokhov: Note that this doesn't link dynamic dependencies, so
      targets produced dynamically will be unavailable. *)
-  link_deps t ~mode ~deps;
-  t
+  link_deps t ~mode ~patch_back_source_tree ~deps;
+  if patch_back_source_tree then (
```

Comment:
I got confused here. Aren't we going to put exactly the set `deps` into the `snapshot`? If yes, why do we need recursive tree traversal?

Reply:
Because the code is simpler this way. We can use the same code twice, i.e. `snapshot t`, rather than traverse `deps` in one case and call `snapshot t` afterwards.

I'm hoping this is a temporary feature, so I favoured code simplicity over performance. But we can revisit if it is a problem in practice.

Reply:
Oh, indeed, that makes sense. Running the same `snapshot` seems better. Maybe worth adding a brief comment clarifying that we expect the first call to `snapshot` to collect exactly `deps`.

Reply:
I added a comment